packages/flutter_tools/lib/src/windows/visual_studio.dart

```diff
@@ -282,6 +283,10 @@ class VisualStudio {
         '-utf8',
         '-latest',
       ];
+      // Ideally this would use Flutter's utf8 encoding which errors on malformed UTF-8.
+      // However, vswhere.exe is known to output unicode replacement characters.
+      // These replacement characters will be ignored unless they affect used properties.
+      // See: https://github.com/flutter/flutter/issues/102451
```

Comment:
Consider moving this up to the `import` line, since that's the line people are most likely to instinctively revert if they spot it.

Reply:
I added a parameter to Flutter's `Encoding` to disable error reporting - I think that makes it clear we're intentionally opting out, thereby making it less likely someone will revert this behavior. I think that addresses your concern but let me know if you have follow-up suggestions!

Reply:
Even better. Thanks!